### PR TITLE
fix: Downgrade rust to 1.70

### PR DESCRIPTION
**Pull request by exdx**
Issues running v1.71 on M1 macs. We can downgrade for now.

Signed-off-by: Dan Sover <dan.sover@avalabs.org>

### CODE REVIEW

**Review by hexfusion — APPROVED** [review]

File: avalanche-network-runner-sdk/Cargo.toml
Comment (hexfusion) [inline avalanche-network-runner-sdk/Cargo.toml:3]: your going to need to push a new crate version for this to work right?

Comment (exdx) [reply avalanche-network-runner-sdk/Cargo.toml:3]: Yes, that's a good point. Since I'm not a maintainer I will wait until @gyuho can take a look.

**Review by gyuho — APPROVED** [review]

**Comment by richardpringle** [comment]
@exdx, patches come out pretty often. I think the better process is to bump the rust version in CI for a time before switching the minimum version of Rust on the crate. 

Also, just because you're working with 1.71 for example, doesn't mean that you're using any features from the lastest release and therefore the minimum value isn't accurate, it was actually some previous version.

**Comment by exdx** [comment]
> @exdx, patches come out pretty often. I think the better process is to bump the rust version in CI for a time before switching the minimum version of Rust on the crate.
> 
> Also, just because you're working with 1.71 for example, doesn't mean that you're using any features from the lastest release and therefore the minimum value isn't accurate, it was actually some previous version.

I agree, I see 1.71.1 was already released https://blog.rust-lang.org/2023/08/03/Rust-1.71.1.html. You recommend moving to that version in CI first?

I agree with your second point.

The reason 1.71 caused issues is a bit unclear to me, it seemed like the toolchain was corrupted in 1.71, at least via rustup. I'm not sure there were any bugs in 1.71 itself. I'm inclined to continue to move to 1.71.1. 

The issue is that by keeping 1.71 here we force any projects that have dependencies on this crate (like avalanche-rust) to also move to 1.71. Which is the latest Rust version. I'm not sure we necessarily want to do this?

**Comment by exdx** [comment]
For now, I need to downgrade because I can't build avalanche-rust locally on 1.70. Specifically I can't build the cargo docs. I think downgrading in this situation is warranted. I don't think we have any upstream guarantees on this crate (or any dependencies really). In the future we can revisit.

**Comment by richardpringle** [comment]
>The issue is that by keeping 1.71 here we force any projects that have dependencies on this crate (like avalanche-rust) to also move to 1.71. Which is the latest Rust version. I'm not sure we necessarily want to do this?

I don't think we want this... unless we're using new features. 

>I agree, I see 1.71.1 was already released https://blog.rust-lang.org/2023/08/03/Rust-1.71.1.html. You recommend moving to that version in CI first?

I take back what I said about bumping CI. What you should be able to do is keep CI at the oldest version that works and use the latest version of Rust locally. The `edition` _should_ guarantee compatibility between dependencies. 

Only when a dependency's min-version increases, or we want to use one of the newer features, should we bump CI and the min-version in our `Cargo.toml`.

Summary:

Keep CI at the oldest possible version and make sure `Cargo.toml` matches that. Only when CI breaks (because you've bumped your local version and are using a new feature) do you need to update both. 

And it might even make sense to do a matrix CI build with the min-version and newest version that only run after code is merged. It's easy to fix things before releasing.

**Pull request merged**
